lib/IDE/ArgumentCompletion.cpp

```diff
@@ -126,6 +126,18 @@ static bool isExpressionResultTypeUnconstrained(const Solution &S, Expr *E) {
   }
 }
 
+/// Returns whether `E` has a parent expression with arguments.
+static bool hasParentCallLikeExpr(Expr *E, ConstraintSystem &CS) {
+  E = CS.getParentExpr(E);
+  while (E) {
+    if (E->getArgs()) {
```

Comment:
I came up with other cases
```swift
let _ = [
  foo: bar(#HERE,
  baz: 1
]
```
Same for tuples. Does it make sense to add check for `TupleExpr`/`ArrayExpr`/`DictionaryExpr`?
Actually, it's not only expressions,  any comma separated syntax might be affected e.g.
```swift
if let foo = bar(#^HERE^#, foo > baz { ...
```
But I think it's okay to ignore them for now.

Reply:
Oh, good catch. I added those cases. I also think it’s OK to ignore the `if` conditions etc for now. We provide argument completions still, so the fallback if we don’t offer the pattern isn’t terrible.